Automattic/syntaxhighlighter
# Add syntax highlighting support for Swift

Based on the initial work by @astralbodies https://github.com/Automattic/syntaxhighlighter/pull/165

Regex and JavaScript aren't my strong suits, please feel free to correct/suggest anything - I just really wanted to see Swift code in p2 😄 

### Changes proposed in this Pull Request

* Add support for Swift based on the following references:
  * [Swift brush from `syntaxhighlighter` repo](https://github.com/syntaxhighlighter/brush-swift/blob/master/brush.js) 6 years ago
  * A bunch of keywords/attributes were added to Swift in the past few years, and I added the commonly used ones with reference from the [official Swift documentation](https://docs.swift.org/swift-book/ReferenceManual/AboutTheLanguageReference.html)
  * Swift implementation in two popular JS libraries:
    * [Prism.js](https://github.com/PrismJS/prism/blob/master/components/prism-swift.js)
    * [Highlight.js](https://github.com/highlightjs/highlight.js/blob/main/src/languages/swift.js)

Please note that there are still some edge cases that the script doesn't cover, like multiline strings with [extended string delimiters](https://docs.swift.org/swift-book/LanguageGuide/StringsAndCharacters.html). I think we can support the most common Swift code usage then add/update more upon feature requests.

### Testing instructions

1. Add or edit a post in Gutenberg editor
1. Add SyntaxHighlighter Code block
1. Select Swift as a language
1. Copy and paste some code from [this gist](https://gist.github.com/jaclync/b201c5d52129a78d500c277af200e12c) or any other Swift examples
1. Save and preview
1. Make sure its syntax highlighted



### Screenshot / Video

You can view some [sample posts on my test site](https://funtestingusa.wpcomstaging.com/tag/swift/).

Gutenberg editor | post
-- | --
<img width="738" alt="Screen Shot 2022-03-14 at 3 38 45 PM" src="https://user-images.githubusercontent.com/1945542/158126023-0676188d-c6b8-4f51-a9dd-e21c127c5aac.png"> | <img width="646" alt="Screen Shot 2022-03-14 at 3 37 54 PM" src="https://user-images.githubusercontent.com/1945542/158125996-601efca5-5899-4018-9b8b-17d3be3d6202.png">
